### DISCUSSION TITLE
[RFC] Quick issue creation

### DISCUSSION THREAD

**Discussion by Wxh16144** [ant-design/ant-design#39870] [RFCs]
## Summary

使用 [Github 创建问题摸板](https://docs.github.com/en/communities/using-templates-to-encourage-useful-issues-and-pull-requests/syntax-for-issue-forms) 代替 https://new-issue.ant.design/。

## Motivation

**方案一**

现目前 http://new-issue.ant.design/ 创建 issue 时，表单必填项太多，过于繁琐。
例如 antd version、 React version、 System info 、 Browser 等必填项，可以建议反馈者通过 [envinfo CLI](https://github.com/tabrindle/envinfo) 快速获取。

`npx envinfo --system --npmPackages '{react*,antd}' --binaries --browsers`

结果:

```
  System:
    OS: macOS 13.1
    CPU: (10) arm64 Apple M1 Pro
    Memory: 164.67 MB / 32.00 GB
    Shell: 5.8.1 - /bin/zsh
  Binaries:
    Node: 16.13.0 - ~/.nvm/versions/node/v16.13.0/bin/node
    Yarn: 1.22.17 - ~/.nvm/versions/node/v16.13.0/bin/yarn
    npm: 8.12.1 - ~/.nvm/versions/node/v16.13.0/bin/npm
  Browsers:
    Chrome: 108.0.5359.124
    Safari: 16.2
  npmPackages:
    antd: 4.21.7 => 4.21.7 
    react: ^17.0.2 => 17.0.2 
    react-dom: ^17.0.2 => 17.0.2 
```

我们可以从 `envinfo` 输出结果得知更多的信息，包括 npm 依赖

## Drawbacks

1. 输入title 后无法检索类似问题。
2. 无法支持多语言。
3. envinfo 对于非主流浏览器(国内二封)的信息可能不准确。

**Comment by afc163** [comment]
可以封装一个 cli 工具，在项目下跑，自动发起 new-issue.ant.design 表单。

1. `npx @ant-design/create-new-issue`
2. 自动获取相关信息，并打印出来作为完整诊断信息。
3. 打开 new-issue.ant.design 并从 query 从带过来预填到表单里。
4. 浏览器版本可以在前端运行时拿到并获取。

**Reply by zombieJ** [reply]
如果是 npx 感觉都可以直接加个 script 到 antd 的包里？

**Reply by JarvisArt** [reply]
> 如果是 npx 感觉都可以直接加个 script 到 antd 的包里？

+1，这样子方便一点，不用安装额外的包

**Reply by afc163** [reply]
项目里万一没 antd 怎么办？或者不在 node_modules/antd 这一级。

**Reply by JarvisArt** [reply]
一般想要提 Issues 都是在开发过程中遇到了问题。
很少说遇到了问题，关闭项目还过了一段时间才去提 Issues，这样他们不太好写出复现步骤。
